compiler/rustc_metadata/src/rmeta/decoder.rs

```diff
@@ -750,12 +751,7 @@ impl MetadataBlob {
     ) -> io::Result<()> {
         let root = self.get_root();
 
-        let all_ls_kinds = vec![
-            "root".to_owned(),
-            "lang_items".to_owned(),
-            "features".to_owned(),
-            "items".to_owned(),
-        ];
+        let all_ls_kinds = ["root", "lang_items", "features", "items"].map(str::to_string);
         let ls_kinds = if ls_kinds.contains(&"all".to_owned()) { &all_ls_kinds } else { ls_kinds };
```

Comment:
Might need some massaging for lifetimes but I think this could be slightly improved:

```suggestion
        let all_ls_kinds = ["root", "lang_items", "features", "items"];
        let ls_kinds = if ls_kinds.iter().any(|name| name == "all") { // avoid the allocation in `.contains()`
            all_ls_kinds.as_slice()
        } else {
            // Allocating one vec on this side will probably be faster than allocating 4 strings
            // Technically don't even need to collect if you can get the iterators working right with
            // `Either` or `&dyn Iterator`, but that is usually a pain
            &ls_kinds.iter().map(|s| s.as_str()).collect::<Vec<&str>>();
        };
```

Reply:
I know, but again, this is out of scope for this PR which is meant to minimally change code structure/implementation to just replace str::to_string and str::to_owned calls.